runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -328,6 +328,22 @@ public void close() throws Exception {
         }
       }
     }
+
+    if (currentOutputWatermark != Long.MAX_VALUE) {
```

Comment:
In non-tests settings, will close() ever be called before outputWatermark reaching to +infinity? For example, when Flink Runner users pause/reload a pipeline?